### PR TITLE
[qgsquick] synchronize map settings and map canvas

### PR DESCRIPTION
**Pull request by 3nids**
this will require new APK delivered to have proper scale bar units (see https://github.com/qgis/QGIS/pull/8059)

### CODE REVIEW

File: src/qml/TransformedPositionSource.qml [hunk 1]
```diff
@@ -11,7 +11,7 @@ PositionSource {
   property CoordinateTransformer ct: CoordinateTransformer {
     id: _ct
     sourceCrs: CrsFactory.fromEpsgId(4326)
-    sourcePosition: _pos.coordinate
+    sourcePosition: Utils.coordinateToPoint(_pos.coordinate)
```
Comment (3nids) [inline src/qml/TransformedPositionSource.qml]: @m-kuhn that's the key change of the whole PR here

File: src/core/qgismobileapp.cpp [hunk 2]
```diff
@@ -99,6 +101,7 @@ QgisMobileapp::QgisMobileapp( QgsApplication *app, QObject *parent )
   connect( this, &QQmlApplicationEngine::quit, app, &QgsApplication::quit );
 
   QgsQuickMapCanvasMap *mMapCanvas = rootObjects().first()->findChild<QgsQuickMapCanvasMap *>();
+  mMapCanvas->mapSettings()->setProject( mProject );
```
Comment (3nids) [inline src/core/qgismobileapp.cpp]: @m-kuhn now that QgsQuickMapSettings has a setProject method, I used it here (successfully tested)

File: src/core/qgismobileapp.cpp [hunk 3]
```diff
@@ -143,13 +145,19 @@ void QgisMobileapp::initDeclarative()
   qRegisterMetaType<QgsRelation>( "QgsRelation" );
 
 
+  // Register QgsQuick QML types
+  qmlRegisterType<QgsQuickMapCanvasMap>( "org.qgis", 1, 0, "MapCanvasMap" );
+  qmlRegisterType<QgsQuickMapSettings>( "org.qgis", 1, 0, "MapSettings" );
+  qmlRegisterType<QgsQuickCoordinateTransformer>( "org.qfield", 1, 0, "CoordinateTransformer" );
+  qmlRegisterSingletonType<QgsQuickUtils>( "Utils", 1, 0, "Utils", utilsSingletonProvider );
```
Comment (3nids) [inline src/core/qgismobileapp.cpp]: registering singleton to access static methods
see https://evileg.com/en/post/263/

**Comment by 3nids** [comment]
build is failing on Travis, but logs are truncated all the time. Restared a few times. Googled… No idea....
https://travis-ci.org/opengisch/QField/jobs/434770776#L1298

**Comment by qfield-fairy** [comment]
Uploaded test apks for [armv7](https://opengis.ch/download/qfield/ci-builds/qfield-dev-374-72f247adf1af71b9c8ec2ad9e1d1d555e312bc83-armv7.apk) and [x86](https://opengis.ch/download/qfield/ci-builds/qfield-dev-374-72f247adf1af71b9c8ec2ad9e1d1d555e312bc83-x86.apk)

**Comment by 3nids** [comment]
@m-kuhn Travis' back to green. Let me know if you want to review (mainly read my comments in such case).

**Comment by m-kuhn** [comment]
code looks good to me. but we need a new sdk for this if I read your first comment correctly?

**Comment by 3nids** [comment]
ok, thanks.
Not exactly remembering (it would be the scale issue only).
But, we'll need a new one anyway to finalize the QgsQuickMapSettings sync.
Can you manage to make it?

**Pull request merged**